docs/content/docs/dev/table/materialized-table/statements.md

```diff
@@ -326,6 +328,69 @@ ALTER MATERIALIZED TABLE my_materialized_table REFRESH PARTITION (ds='2024-06-28
 <span class="label label-danger">Note</span>
 - The REFRESH operation will start a Flink batch job to refresh the materialized table data.
 
+## AS <select_statement>
+
+```sql
+ALTER MATERIALIZED TABLE [catalog_name.][db_name.]table_name AS <select_statement>
+```
+
+The `AS <select_statement>` clause allows you to modify the query definition of a materialized table. It updates the query used by the materialized table refresh job and infers a new schema based on the updated query to adjust the table’s schema. However, this operation does not directly affect existing data.
+
+The modification process depends on the refresh mode of the materialized table:
+
+**Full mode:**
```

Comment:
```
**Full Mode:**

1. Update the `schema` and `query definition` of the materialized table.
2. The table is refreshed using the new query definition when the next refresh job is triggered:
   - If it is a partitioned table and [partition.fields.#.date-formatter]({{< ref "docs/dev/table/config" >}}#partition-fields-date-formatter) is correctly set, only the latest partition will be refreshed.
   - Otherwise, the table will be overwritten entirely.
```